src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -61,14 +65,14 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
 
                 if (!Interop.Ssl.Tls13Supported)
                 {
-                    if (protocols != SslProtocols.None &&
+                    if (sslAuthenticationOptions.EnabledSslProtocols != SslProtocols.None &&
                         CipherSuitesPolicyPal.WantsTls13(protocols))
                     {
                         protocols = protocols & (~SslProtocols.Tls13);
                     }
                 }
                 else if (CipherSuitesPolicyPal.WantsTls13(protocols) &&
-                    CipherSuitesPolicyPal.ShouldOptOutOfTls13(sslAuthenticationOptions.CipherSuitesPolicy, policy))
+                    CipherSuitesPolicyPal.ShouldOptOutOfTls13(sslAuthenticationOptions.CipherSuitesPolicy, sslAuthenticationOptions.EncryptionPolicy))
```

Comment:
I assume this object is immutable, or at least not accessible to user code?  Seems like interesting issues could arise if the value could change dynamically out from under us?

Reply:
sslAuthenticationOptions is copy of Client or Server Authentication options and it not user accessible. We basically take a snapshot when `AuthenticateAS*()` is called. 
https://github.com/dotnet/runtime/blob/a4894885a534c3c9091fd7dedae18d2686c6b844/src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs#L35